code/datums/records/manifest.dm

```diff
@@ -0,0 +1,206 @@
+				var/list/_internal = manifest_out[department.dept_id]
+				_internal.Insert(put_at_top, list(entry))
+		else
+			LAZYINITLIST(manifest_out["Misc"])
+			var/put_at_top = (hud == JOB_HUD_CAPTAIN) || (hud == JOB_HUD_ACTINGCAPTAIN) || (heads[rank])
+			var/list/_internal = manifest_out["Misc"]
+			_internal.Insert(put_at_top, list(entry))
+
+	// 'manifest_out' is not sorted.
+	var/list/sorted_out = list()
+	for(var/datum/department_group/department as anything in SSdepartment.sorted_department_for_manifest)
+		if(isnull(manifest_out[department.dept_id]))
+			continue
+		sorted_out[department.manifest_category_name] = manifest_out[department.dept_id] // this also changes a department name.
+	return sorted_out
+
+/// Returns the manifest as an html.
+/datum/manifest/proc/get_html(monochrome = FALSE)
+	var/list/manifest = get_manifest()
+	var/dat = {"
+	<head><style>
+		.manifest {border-collapse:collapse;}
+		.manifest td, th {border:1px solid [monochrome?"black":"#DEF; background-color:white; color:black"]; padding:.25em}
+		.manifest th {height: 2em; [monochrome?"border-top-width: 3px":"background-color: #48C; color:white"]}
+		.manifest tr.head th { [monochrome?"border-top-width: 1px":"background-color: #488;"] }
+		.manifest tr.alt td {[monochrome?"border-top-width: 2px":"background-color: #DEF"]}
+	</style></head>
+	<table class="manifest" width='350px'>
+	<tr class='head'><th>Name</th><th>Rank</th></tr>
+	"}
+	for(var/department in manifest)
+		var/list/entries = manifest[department]
+		dat += "<tr><th colspan=3>[department]</th></tr>"
+		//JUST
+		var/even = FALSE
+		for(var/entry in entries)
+			var/list/entry_list = entry
+			dat += "<tr[even ? " class='alt'" : ""]><td>[entry_list["name"]]</td><td>[entry_list["rank"]]</td></tr>"
+			even = !even
+
+	dat += "</table>"
+	dat = replacetext(dat, "\n", "")
+	dat = replacetext(dat, "\t", "")
+	return dat
+
+/datum/manifest/proc/inject(mob/living/carbon/human/person)
+	set waitfor = FALSE
+
+	// We need to compile the overlays now, otherwise we're basically copying an empty icon.
+	COMPILE_OVERLAYS(person)
+	var/mutable_appearance/character_appearance = new(person.appearance)
+	var/datum/dna/stored/record_dna = new()
+	person.dna.copy_dna(record_dna)
+	var/person_gender = "Other"
+	if(person.gender == "male")
+		person_gender = "Male"
+	if(person.gender == "female")
+		person_gender = "Female"
+	var/assignment = person.mind.assigned_role
+	var/datum/bank_account/bank_account = person.get_bank_account()
+
+	var/datum/record/locked/lockfile = new(
+		age = person.age,
+		blood_type = record_dna.blood_type,
+		character_appearance = character_appearance,
+		dna_string = record_dna.unique_enzymes,
+		fingerprint = md5(record_dna.uni_identity),
+		gender = person_gender,
+		initial_rank = assignment,
+		name = person.real_name,
+		rank = assignment,
+		species = record_dna.species.name,
+		// Locked specifics
+		dna_ref = record_dna,
+		mind_ref = person.mind,
```

Comment:
these are not ref. these are direct access to the instance.
It should be either having a proper var name (removing `_ref`) or doing `FAST_REF(thing)`